ext/standard/head.c

```diff
@@ -294,39 +294,34 @@ PHP_FUNCTION(setrawcookie)
 /* {{{ Returns true if headers have already been sent, false otherwise */
 PHP_FUNCTION(headers_sent)
 {
-	zval *arg1 = NULL, *arg2 = NULL;
-	const char *file="";
-	int line=0;
+	zval *by_ref_filename = NULL;
+	zval *by_ref_line = NULL;
+	const char *file = "";
+	int line = 0;
 
 	ZEND_PARSE_PARAMETERS_START(0, 2)
 		Z_PARAM_OPTIONAL
-		Z_PARAM_ZVAL(arg1)
-		Z_PARAM_ZVAL(arg2)
+		Z_PARAM_ZVAL(by_ref_filename)
+		Z_PARAM_ZVAL(by_ref_line)
 	ZEND_PARSE_PARAMETERS_END();
 
 	if (SG(headers_sent)) {
 		line = php_output_get_start_lineno();
 		file = php_output_get_start_filename();
 	}
 
-	switch(ZEND_NUM_ARGS()) {
```

Comment:
I think the code was written this way because this avoids a check: if we have a second param then we know the first one was also passed. The new code doesn't avoid such a check. It probably doesn't really matter however.